pkgs/development/haskell-modules/configuration-ghc-9.10.x.nix

```diff
@@ -14,7 +16,7 @@ self: super: {
   base = null;
   binary = null;
   bytestring = null;
-  Cabal = null;
+  Cabal = doDistribute self.Cabal_3_12_1_0; # GHC ships with Cabal 3.12.0.0, but there is no corresponding cabal-install
```

Comment:
We shouldn’t override Cabal globally but only for those packages which can’t build with 3.12.0.

cabal-install already has an override to use Cabal_3_12_1_0 in configuration-common.nix, I think. Why isn’t that enough?

Reply:
Oh, that's a much better idea. I needed it for `cabal-install-solver`.

Looks like these overrides in `configuration-common.nix` only apply if we're working with GHC <= 9.10. When I apply them to GHC 9.10.1 too, it seems to fix the issue so we don't need this `Cabal` override.

Any objection to me removing this line?

https://github.com/NixOS/nixpkgs/blob/5de1564aed415bf9d0f281461babc2d101dd49ff/pkgs/development/haskell-modules/configuration-common.nix#L55

Reply:
I changed this line to refer to `self.ghc.version "9.10.2"`.